plugins/CoreHome/vue/src/MatomoUrl/MatomoUrl.ts

```diff
@@ -54,7 +54,23 @@ class MatomoUrl {
     this.url.value = new URL(window.location.href);
 
     window.addEventListener('hashchange', (event) => {
-      this.url.value = new URL(event.newURL);
+      const newUrl = new URL(event.newURL);
+
+      // The URL hash part is used to store the currently shown location
+      // However, in some cases, like the skip to content link, we want a hash link to act like a
+      // jump-to mark. Therefor we first check if the new hash looks like an anchor/id to jump to
+      // if an element with that id / link with that name exists, we scroll it into view and reset
+      // the changed hash in the url.
+      if (
+        newUrl.hash.match(/^#[a-z0-9_-]+$/i)
+        && $(`${newUrl.hash},a[name=${newUrl.hash.substring(1)}]`).length
+      ) {
+        $(`${newUrl.hash},a[name=${newUrl.hash.substring(1)}]`)[0].scrollIntoView();
+        window.location.href = event.oldURL;
```

Comment:
Can this have some really ugly side effects? :thinking: 

For example the inline link for image tracking in the javascript tracking code generator:

https://github.com/matomo-org/matomo/blob/9a3ef94df61682e7d543f3f8b438c368f6eb71a6/plugins/CoreAdminHome/vue/src/JsTrackingCodeGenerator/JsTrackingCodeGenerator.vue#L213-L219

The actual section links at the top are always going to `#/section`, not just `#section`. That one inline link now goes to the actual anchor, and if you have any other section in your hash, like `#/react`, the change back to the previous hash will also trigger a scroll to the previous hash anchor.

Now this is a simple one to fix (just update the inline anchor link), but maybe there are more places that would actually break when cycling hashes?

Reply:
Would it make sense to check if the previous hash was something containing `=` or `&` and only do it in that case? 🤔

Reply:
Ideally I would say we replace the skip-to-content link with a pure JS implementation: focus the main element, scroll that element into view, don't use any hash manipulation.

If that is sufficient for accessibility we wouldn't need to change any hashes around :thinking:

Reply:
I would probably also vote for not updating the hash in the actual URL in browser and just do the scrolling via JS.